Core/MaterialExtension.cpp

```diff
@@ -16,66 +16,103 @@
 #include "MaterialParameters.h"
 #include "MaterialExtension.h"
 
-MaterialExtension::MaterialExtension(const G4String& name, const G4Material* baseMaterial) :
+MaterialExtension::MaterialExtension( G4String materialType, const G4String& name, const G4Material* baseMaterial) :
   G4Material(
     name, baseMaterial->GetDensity(), baseMaterial, baseMaterial->GetState(),
     baseMaterial->GetTemperature(), baseMaterial->GetPressure()
   )
 {
-  foPslifetime = 0.;
-  foPsFraction = 0.;
   fTarget = false ;
   //! depends on the material typ
-  fPickOffFraction = 0.;
+  Material = new MaterialParameters();
+  Material->SetMaterialByName(materialType);
+  FillIntensities();
 }
 
-
 MaterialExtension::~MaterialExtension() {}
 
-const G4double MaterialExtension::Get3gFraction()
+void MaterialExtension::Add_oPsComponent( G4double Lifetime, G4double Probability )
 {
-  return foPsFraction;
+  if(Lifetime < 0. || Lifetime > 142.) 
```

Comment:
I would recommend not to put any hardcoded variables (why 142? what units?) in here and (optionally) follow the convention of naming variables with lower-case letters (lifetime instead of Lifetime)

Reply:
Values changed to parameters of MaterialParameters class